Add kafka sink and source 3.1.0.4 doc (#321)
streamnative/pulsar-hub: connectors/kafka-source/3.1.0.4/kafka-source.md

icon: "/images/connectors/kafka-logo.png"
download: "https://github.com/streamnative/pulsar/releases/download/v3.1.0.4/pulsar-io-kafka-3.1.0.4.nar"
support: StreamNative
support_link: https://streamnative.io
support_img: "/images/streamnative.png"
owner_name: ""
owner_img: ""
dockerfile: "https://hub.docker.com/r/streamnative/pulsar-io-kafka"
sn_available: true
id: "kafka-source"
---

The [Kafka](https://kafka.apache.org/) source connector pulls messages from Kafka topics and persists the messages to Pulsar topics. For more information about connectors, see [Connector Overview](https://docs.streamnative.io/docs/connector-overview).

This document introduces how to get started with creating a Kafka source connector and get it up and running.

![](/images/connectors/kafka-source.png)

## Quick start

### Prerequisites

The prerequisites for connecting an Kafka source connector to external systems include:

Apache Kafka: Ensure you have a running Kafka instance. You can follow the official Kafka [Quickstart guide](https://kafka.apache.org/quickstart) to set up a Kafka instance if you don't have one already.

### 1. Create a connector

> Depending on the environment, there are several ways to create an Kafka sink connector:

Reviewer: ```suggestion
Depending on the environment, there are several ways to create a Kafka source connector:
```

- [Create a Connector on StreamNative Cloud](https://docs.streamnative.io/docs/connector-create).
- [Create a Connector with Function worker](https://pulsar.apache.org/docs/io-quickstart/).
Using this way requires you to download a **NAR** package to create a connector. You can download the version you need from the `download button` at the beginning of the article.
- [Create a Connector with Function mesh](https://functionmesh.io/docs/connectors/run-connector).
Using this way requires you to set the docker image. You can choose the version you want to launch from [here](https://hub.docker.com/r/streamnative/pulsar-io-elastic-search).

> No matter how you create an Kafka sink connector, the minimum connector configuration contains the following parameters:

Reviewer: ```suggestion
No matter how you create a Kafka source connector, the minimum connector configuration contains the following parameters:
```
